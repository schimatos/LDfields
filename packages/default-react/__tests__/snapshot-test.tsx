import React from 'react';
import renderer from 'react-test-renderer';
import { axe, toHaveNoViolations } from 'jest-axe';
import { literal, namedNode, quad } from '@rdfjs/data-model';
import Input from '../src';

expect.extend(toHaveNoViolations);

describe('Testing the snapshot changes of fields', () => {
  it('Should have no violations on the default render for empty props', async () => {
    const tree = renderer.create(
      <Input
        onChange={() => {}}
        constraints={{}}
        props={{
          term: undefined,
          annotations: [],
        }}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should have no violations on the default render for namedNode props', async () => {
    const tree = renderer.create(
      <Input
        onChange={() => {}}
        constraints={{}}
        props={{
          term: namedNode('http://example.org#Jesse'),
          annotations: [],
        }}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should have no violations on the default render for namedNode props + label', async () => {
    const tree = renderer.create(
      <Input
        onChange={() => {}}
        constraints={{}}
        props={{
          term: namedNode('http://example.org#Jesse'),
          annotations: [
            quad(
              namedNode('http://example.org#Jesse'),
              namedNode('http://www.w3.org/1999/02/22-rdf-syntax-ns#type'),
              literal('Jesse'),
            ),
          ],
        }}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should have no violations on the default render for literal', async () => {
    const tree = renderer.create(
      <Input
        onChange={() => {}}
        constraints={{}}
        props={{
          term: literal('Jesse'),
          annotations: [],
        }}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should have no violations on the default render for integer literal', async () => {
    const tree = renderer.create(
      <Input
        onChange={() => {}}
        constraints={{}}
        props={{
          term: literal('0', 'http://www.w3.org/2001/XMLSchema#integer'),
          annotations: [],
        }}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
