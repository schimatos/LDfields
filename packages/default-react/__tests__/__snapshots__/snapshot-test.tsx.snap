// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the snapshot changes of fields Should have no violations on the default render for empty props 1`] = `
<fieldset
  onBlur={[Function]}
>
  <label>
    value
  </label>
  <input
    aria-label="value"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  />
  <select
    aria-label="termType"
    onChange={[Function]}
    value="Literal"
  >
    <option
      value="NamedNode"
    >
      NamedNode
    </option>
    <option
      value="BlankNode"
    >
      BlankNode
    </option>
    <option
      value="Literal"
    >
      Literal
    </option>
  </select>
</fieldset>
`;

exports[`Testing the snapshot changes of fields Should have no violations on the default render for integer literal 1`] = `
<fieldset
  onBlur={[Function]}
>
  <label>
    value
  </label>
  <input
    aria-label="value"
    onBlur={[Function]}
    onChange={[Function]}
    value="0"
  />
  <select
    aria-label="termType"
    onChange={[Function]}
    value="Literal"
  >
    <option
      value="NamedNode"
    >
      NamedNode
    </option>
    <option
      value="BlankNode"
    >
      BlankNode
    </option>
    <option
      value="Literal"
    >
      Literal
    </option>
  </select>
  <label>
    datatype
  </label>
  <input
    aria-label="datatype"
    onBlur={[Function]}
    onChange={[Function]}
    value="http://www.w3.org/2001/XMLSchema#integer"
  />
</fieldset>
`;

exports[`Testing the snapshot changes of fields Should have no violations on the default render for literal 1`] = `
<fieldset
  onBlur={[Function]}
>
  <input
    aria-label="value"
    onChange={[Function]}
    type="text"
    value="Jesse"
  />
  <select
    aria-label="termType"
    onChange={[Function]}
    value="Literal"
  >
    <option
      value="NamedNode"
    >
      NamedNode
    </option>
    <option
      value="BlankNode"
    >
      BlankNode
    </option>
    <option
      value="Literal"
    >
      Literal
    </option>
  </select>
  <label>
    datatype
  </label>
  <input
    aria-label="datatype"
    onBlur={[Function]}
    onChange={[Function]}
    value="http://www.w3.org/2001/XMLSchema#string"
  />
</fieldset>
`;

exports[`Testing the snapshot changes of fields Should have no violations on the default render for namedNode props + label 1`] = `
<fieldset
  onBlur={[Function]}
>
  <label>
    value
  </label>
  <input
    aria-label="value"
    onBlur={[Function]}
    onChange={[Function]}
    value="http://example.org#Jesse"
  />
  <select
    aria-label="termType"
    onChange={[Function]}
    value="NamedNode"
  >
    <option
      value="NamedNode"
    >
      NamedNode
    </option>
    <option
      value="BlankNode"
    >
      BlankNode
    </option>
    <option
      value="Literal"
    >
      Literal
    </option>
  </select>
  <label>
    type
  </label>
  <input
    aria-label="type"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  />
  <label>
    label
  </label>
  <input
    aria-label="label"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  />
</fieldset>
`;

exports[`Testing the snapshot changes of fields Should have no violations on the default render for namedNode props 1`] = `
<fieldset
  onBlur={[Function]}
>
  <label>
    value
  </label>
  <input
    aria-label="value"
    onBlur={[Function]}
    onChange={[Function]}
    value="http://example.org#Jesse"
  />
  <select
    aria-label="termType"
    onChange={[Function]}
    value="NamedNode"
  >
    <option
      value="NamedNode"
    >
      NamedNode
    </option>
    <option
      value="BlankNode"
    >
      BlankNode
    </option>
    <option
      value="Literal"
    >
      Literal
    </option>
  </select>
  <label>
    type
  </label>
  <input
    aria-label="type"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  />
  <label>
    label
  </label>
  <input
    aria-label="label"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  />
</fieldset>
`;
